zerver/lib/narrow.py

```diff
@@ -84,7 +84,7 @@ def narrow_filter(event: Mapping[str, Any]) -> bool:
             elif operator == "is" and operand == "private":
                 if message["type"] != "private":
                     return False
-            elif operator == "is" and operand in ["starred"]:
+            elif operator == "is" and operand == "starred":
                 if operand not in flags:
                     return False
```

Comment:
This logic intentionally used the `in` style because that would support additional flags that work like `starred` more easily.